### PR TITLE
statistics: log the db name

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/pull/48929#discussion_r1407975768

Problem Summary:

### What changed and how does it work?

Printed the DB name in the logs.

Because changing the interface would have a significant invasive impact, the ctx was attached to the DDL handler. Otherwise, not only all the tests but also the handle's flush operation would have to query the schema by themselves.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/ddl/ddl.go [hunk 1]
```diff
@@ -125,9 +130,12 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 			// Get the count and modify count of the partition.
 			stats := h.statsHandler.GetPartitionStats(globalTableInfo, def.ID)
 			if stats.Pseudo {
+				is := h.ctx.GetDomainInfoSchema().(infoschema.InfoSchema)
```
Comment (hi-rustin) [inline pkg/statistics/handle/ddl/ddl.go:133]: Usually, this won't happen, so putting the getting scheme operation inside the if statement is OK.

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Comment by hi-rustin** [comment]
Tested again:
```log
[2023/11/29 19:16:03.458 +08:00] [WARN] [ddl.go:153] ["drop partition with pseudo stats, usually it won't happen because we always load stats when initializing the handle"] [category=stats] [schema=test] [table=pt] [partition=p0]
```

**Review by time-and-fate — APPROVED** [review]

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Review by qw4990 — APPROVED** [review]

**Pull request merged**